core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/col.kt

```diff
@@ -0,0 +1,585 @@
+package org.jetbrains.kotlinx.dataframe.api
+
+import org.jetbrains.kotlinx.dataframe.AnyColumnGroupAccessor
+import org.jetbrains.kotlinx.dataframe.ColumnGroupReference
+import org.jetbrains.kotlinx.dataframe.DataFrame
+import org.jetbrains.kotlinx.dataframe.DataRow
+import org.jetbrains.kotlinx.dataframe.columns.ColumnAccessor
+import org.jetbrains.kotlinx.dataframe.columns.ColumnGroup
+import org.jetbrains.kotlinx.dataframe.columns.ColumnPath
+import org.jetbrains.kotlinx.dataframe.columns.ColumnSet
+import org.jetbrains.kotlinx.dataframe.columns.SingleColumn
+import org.jetbrains.kotlinx.dataframe.documentation.LineBreak
+import org.jetbrains.kotlinx.dataframe.impl.columns.getAt
+import org.jetbrains.kotlinx.dataframe.impl.columns.singleImpl
+import org.jetbrains.kotlinx.dataframe.impl.columns.transformSingle
+import org.jetbrains.kotlinx.dataframe.util.COL_REPLACE
+import org.jetbrains.kotlinx.dataframe.util.IDENTITY_FUNCTION
+import kotlin.reflect.KProperty
+
+// region ColumnsSelectionDsl
+
+public interface ColColumnsSelectionDsl {
+
+    /**
+     * ## Col
+     *
+     * Creates a [ColumnAccessor] (or [SingleColumn]) for a column with the given argument which can be either
+     * an index ([Int]) or a reference to a column ([String], [ColumnPath], [KProperty], or [ColumnAccessor]).
+     *
+     * This is a DSL-shorthand for [column] and can be both typed and untyped (in case you're supplying
+     * a column name, -path, or index).
+     * The function can also be called on [ColumnGroups][ColumnGroupReference] to create
+     * an accessor for a column inside a [ColumnGroup].
+     * {@include [LineBreak]}
+     * {@includeArg [CommonColDocs.Note]}
+     *
+     * #### For example:
+     *
+     * `df.`[select][DataFrame.select]` { `[col][col]`<`[String][String]`>("colA") }`
+     *
+     * `df.`[select][DataFrame.select]` { `[col][col]`(SomeType::colB) }`
+     *
+     * `df.`[select][DataFrame.select]` { myColumnGroup.`[col][col]`(1) }`
+     *
+     * #### Examples for this overload:
+     *
+     * {@includeArg [CommonColDocs.ExampleArg]}
+     *
+     * To create a [ColumnAccessor] for a specific kind of column with runtime checks, take a look at the functions
+     * [valueCol][ColumnsSelectionDsl.valueCol],
+     * [colGroup][ColumnsSelectionDsl.colGroup],
+     * and [frameCol][ColumnsSelectionDsl.frameCol].
+     *
+     * @return A [ColumnAccessor] for the column with the given argument if possible, else a [SingleColumn].
+     * @throws [IllegalStateException\] if the column with the given argument does not exist.
+     *
+     * @see [column\]
+     * @see [ColumnsSelectionDsl.colGroup\]
+     * @see [ColumnsSelectionDsl.frameCol\]
+     * @see [ColumnsSelectionDsl.valueCol\]
+     * {@arg [CommonColDocs.Note]}
+     */
+    private interface CommonColDocs {
```

Comment:
Is it correct to say, that this is a util documentation interface. If it is, probably better to move outside this file

Reply:
I don't think it's worth it to clutter the (internal) naming space with documentation that only applies to this file. New inheritance support could help a tiny bit with that, though.
Although, if it's in a separate file, updating even a single letter of documentation will create a git diff of 4 files (2 files in the project and 2 generated ones).

Reply:
plus, this documentation describes the functions inside this interface. Usually, kdoc is always right next to the function, so it should probably still be close to the function